cluster_experiments/power_analysis.py

```diff
@@ -428,99 +580,73 @@ def from_config(cls, config: PowerConfig) -> "PowerAnalysis":
-        assert (
-            self.analysis.treatment in self.splitter.treatments
-        ), f"treatment in analysis ({self.analysis.treatment}) must be in treatments in splitter ({self.splitter.treatments})"
-
-        assert (
-            self.control in self.splitter.treatments
-        ), f"control in power analysis ({self.control}) must be in treatments in splitter ({self.splitter.treatments})"
-
-    def check_covariates(self):
-        if hasattr(self.analysis, "covariates"):
-            cupac_in_covariates = (
-                self.cupac_handler.cupac_outcome_name in self.analysis.covariates
-            )
-
-            assert cupac_in_covariates or not self.cupac_handler.is_cupac, (
-                f"covariates in analysis must contain {self.cupac_handler.cupac_outcome_name} if cupac_model is not None. "
-                f"If you want to use cupac_model, you must add the cupac outcome to the covariates of the analysis "
-                f"You may want to do covariates=['{self.cupac_handler.cupac_outcome_name}'] in your analysis method or your config"
-            )
-
-            if hasattr(self.splitter, "cluster_cols"):
-                if set(self.analysis.covariates).intersection(
-                    set(self.splitter.cluster_cols)
-                ):
-                    logging.warning(
-                        f"covariates in analysis ({self.analysis.covariates}) are also cluster_cols in splitter ({self.splitter.cluster_cols}). "
-                        f"Be specially careful when using switchback splitters, since the time splitter column is being overriden"
-                    )
-
-    def check_clusters(self):
-        has_analysis_clusters = hasattr(self.analysis, "cluster_cols")
-        has_splitter_clusters = hasattr(self.splitter, "cluster_cols")
-        not_cluster_cols_cond = not has_analysis_clusters or not has_splitter_clusters
-        assert (
-            not_cluster_cols_cond
-            or self.analysis.cluster_cols == self.splitter.cluster_cols
-        ), f"cluster_cols in analysis ({self.analysis.cluster_cols}) must be the same as cluster_cols in splitter ({self.splitter.cluster_cols})"
-
-        assert (
-            has_splitter_clusters
-            or not has_analysis_clusters
-            or not self.analysis.cluster_cols
-            or isinstance(self.splitter, RepeatedSampler)
-        ), "analysis has cluster_cols but splitter does not."
-
-        assert (
-            has_analysis_clusters
-            or not has_splitter_clusters
-            or not self.splitter.cluster_cols
-        ), "splitter has cluster_cols but analysis does not."
-
-        has_time_col = hasattr(self.splitter, "time_col")
-        assert not (
-            has_time_col
-            and has_splitter_clusters
-            and self.splitter.time_col not in self.splitter.cluster_cols
-        ), "in switchback splitters, time_col must be in cluster_cols"
-
     def check_inputs(self):
         self.check_covariates()
```

Comment:
here you can call super, no?

Reply:
do you mean using the check_inputs method of super, and then add the additional check_perturbator_consistency() method to overwrite the subclass check_inputs() method?

Reply:
something like:
```python
super.check_inputs()
self.check_perturbator_consistency()
```
though my syntax may be off